src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs

```diff
@@ -864,13 +951,61 @@ public override void ViewModelDetached()
             }
         }
         
-        [ExcludeFromCodeCoverage]
+        ILaunchSettingsProvider _launchSettingsProvider;
         protected virtual ILaunchSettingsProvider GetDebugProfileProvider()
         {
-            return UnconfiguredProject.Services.ExportProvider.GetExportedValue<ILaunchSettingsProvider>();
+            if(_launchSettingsProvider == null)
+            {
+                _launchSettingsProvider = UnconfiguredProject.Services.ExportProvider.GetExportedValue<ILaunchSettingsProvider>();
+            }
+
+            return _launchSettingsProvider;
         }
-                
-        internal class LaunchType
+
+        /// <summary>
+        /// Called by the currently active control when errors within the control have changed
+        /// </summary>
+        private void OnCustomUIErrorsChanged(Object sender, DataErrorsChangedEventArgs e)
+        {
+            ErrorsChanged?.Invoke(this, e);
+
+            OnPropertyChanged(nameof(DoesNotHaveErrors));
+        }
+
+        public IEnumerable GetErrors(string propertyName)
+        {
+            return null;
+        }
+
+        /// <summary>
+        /// We are considered in error if we have errors, or the currently active control has errors
+        /// </summary>
+        public bool HasErrors
+        {
+            get
+            {
+                var notifyDataError = ActiveProvider?.CustomUI?.DataContext as INotifyDataErrorInfo;
+                if(notifyDataError != null)
+                {
+                    if(notifyDataError.HasErrors)
```

Comment:
My original thought was that if we add errors (like convert the way env variable errors hare handled) to the main dialog, it would do its check later where I currently return false. But I fixed this as you suggested.